pid_controller/src/pid_controller.cpp

```diff
@@ -495,6 +517,43 @@ controller_interface::return_type PidController::update_and_write_commands(
     state_interfaces_values_[i] = measured_state_values_[i];
   }
 
+  // Calculate size of DOF references for indexing
+  const size_t dof_reference_size = dof_ * params_.reference_and_state_interfaces.size();
+
+  if (params_.export_params.gain_references)
+  {
+    size_t gains_start_index = dof_reference_size;
+    for (size_t i = 0; i < dof_; ++i)
+    {
+      auto current_pid_gains = pids_[i]->get_gains();
+      for (size_t j = 0; j < GAIN_INTERFACES.size(); ++j)
+      {
+        const size_t buffer_index = gains_start_index + i + j * dof_;
```

Comment:
```suggestion
        const size_t buffer_index = gains_start_index + (j * dof_) + i;
```

As the standard formula for index calculation is,
`index = base_offset + (row_index * row_size) + column_index`.

While the previous one is perfectly fine, this improves general readability and precedence readability.